oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporterTest.java

```diff
@@ -538,4 +557,153 @@ private void dumpIndexDefinitions(String... indexPaths) throws IOException, Comm
         printer.print(pw, Format.JSON, false);
         Files.write(sw.toString(), file, UTF_8);
     }
+
+    private String importData_IncrementalUpdate_Before_Setup_Method() throws IOException, CommitFailedException {
+        NodeBuilder builder = store.getRoot().builder();
+        createIndexDefinition(builder.child(INDEX_DEFINITIONS_NAME),
+                "fooIndex", true, false, ImmutableSet.of("foo"), null)
+                .setProperty(ASYNC_PROPERTY_NAME, "async");
+        builder.child("a").setProperty("foo", "abc");
+        builder.child("b").setProperty("foo", "abc");
+        store.merge(builder, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+
+        new AsyncIndexUpdate("async", store, provider).run();
+
+        String checkpoint = createIndexDirs("/oak:index/fooIndex");
+        builder = store.getRoot().builder();
+        builder.child("c").setProperty("foo", "abc");
+        builder.child("d").setProperty("foo", "abc");
+        store.merge(builder, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+
+        new AsyncIndexUpdate("async", store, provider).run();
+
+        FilterImpl f = createFilter(store.getRoot(), NT_BASE);
+        PropertyIndexLookup lookup = new PropertyIndexLookup(store.getRoot());
+        assertEquals(of("a", "b", "c", "d"), find(lookup, "foo", "abc", f));
+
+        builder = store.getRoot().builder();
+        builder.child("e").setProperty("foo", "abc");
+        store.merge(builder, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+
+        return checkpoint;
+    }
+
+    private IndexImporterProvider getImporterProvider(String checkpoint) {
+        IndexImporterProvider importerProvider = new IndexImporterProvider() {
+            @Override
+            public void importIndex(NodeState root, NodeBuilder defn, File indexDir) {
+                assertEquals("fooIndex", indexDir.getName());
+                assertEquals(2, defn.getProperty(REINDEX_COUNT).getValue(Type.LONG).longValue());
+                defn.getChildNode(IndexConstants.INDEX_CONTENT_NODE_NAME).remove();
+
+                NodeState cpState = store.retrieve(checkpoint);
+                NodeState indexData = NodeStateUtils.getNode(cpState, "/oak:index/fooIndex/:index");
+                defn.setChildNode(IndexConstants.INDEX_CONTENT_NODE_NAME, indexData);
+            }
+
+            @Override
+            public String getType() {
+                return "property";
+            }
+        };
+        return importerProvider;
+    }
+
+    @Test
+    public void importData_IncrementalUpdate_Test_Failure_AT_SWITCHLANE_State() throws Exception {
```

Comment:
This is a mix of camel case, underscore, caps...
importDataIncrementalUpdateTestFailureAtSwitchLaneState